### PR TITLE
Pass number of responses along whenever we return a thread_id

### LINKED ISSUE #1144: Return the number of the replies/comments in the discussion object in the API
Screenshot 2021-04-26 at 03 10 49

In the design above, we want to show the number of replies a discussion has. This discussion card/component is used to in the discussions list page/tab.

Currently there is no way to get this information directly without an additional request to get the thread to figure this out, even when we don't need the rest of the thread's data at this stage.

Also, even if the additional request is not an issue, this is requiring the frontend to do more data wrangling than it should to figure this out, when I think it would be cleaner to do this in the backend instead. Apparently @kalvdans [agrees here](https://github.com/Couchers-org/couchers/pull/578#discussion_r573989349) so maybe this isn't a terrible idea 😉

### PR DESCRIPTION
**Pull request by kalvdans**
Closes https://github.com/Couchers-org/couchers/issues/1144.

**Backend checklist**
- [x] Formatted my code by running `isort . && black .` in `app/backend`
- [x] Added tests for any new code or added a regression test if fixing a bug
- ~~[ ] All tests pass~~ not needed since we are Backend&trade;
- [ ] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

**Comment by aapeliv** [comment]
How about adding a submessage like this:

```proto
message Thread {
  int64 thread_id = 1;
  int64 num_responses = 2;
}
```

Then in `Page`, we could do

```proto
message Page {
  // ...
  Thread thread = 20;
  // ...
}
```

That way we can have a `thread_to_pb` and reuse it everywhere and be able to add any additional things to the thread as well, as needed. In the frontend too, we can then just pass that over to whatever component needs it.

I think that would be cleaner and keep the thread object more modular?

**Comment by kalvdans** [comment]
> How about adding a submessage like this:

Good idea, I'll make the suggested change and see how it looks like.

**Comment by kalvdans** [comment]
Looks nicer @aapeliv ? Also I'm frontend analphabet so need help to adapt the frontend :white_flag:

File: app/pb/threads.proto
Comment (aapeliv) [inline app/pb/threads.proto]: Move this under the service definition.

I would also just call it `Thread` and specify it's top level in a comment

File: app/pb/threads.proto
Comment (aapeliv) [inline app/pb/threads.proto:56]: So `TopLevelThread` returns total number of responses for the recursive thread, and `Reply` contains the num_replies for comments?

**Review by aapeliv — COMMENTED** [review]
Just a few small comments, looks good in general

**Comment by aapeliv** [comment]
Regarding merge: let's leave the frontend as-is and let @darrenvong merge/rebase this into his work. There's no point going to the old discussion stuff and changing that, it'll just cause pain and suffering in merge for the new discussion stuff.

File: app/pb/threads.proto
Comment (kalvdans) [reply app/pb/threads.proto:56]: Yes, exactly. There's no way to get the number of comments (without counting replies to comments), but that could be one thing to be added to TopLevelThread in the future.

Comment (aapeliv) [reply app/pb/threads.proto:56]: That's not a problem, I think. I don't think we need that (# comments without replies).

The main thing was to make sure we return number of replies for each comment as well.

**Comment by kalvdans** [comment]
> Regarding merge: let's leave the frontend as-is and let @darrenvong merge/rebase this into his work. There's no point going to the old discussion stuff and changing that, it'll just cause pain and suffering in merge for the new discussion stuff.

Ok! Life is easy being a backend developer ;)

**Review by aapeliv — DISMISSED** [review]
LGTM!

Let's leave it to @darrenvong to take it from here and integrate into what he's doing.

**Comment by aapeliv** [comment]
> * ~[ ] All tests pass~ not needed since we are Backend™

🤣

**Comment by darrenvong** [comment]
Pending rebuild, the frontend itself and the tests should be fixed now (within this branch) to read from the nested thread object, so you can merge to develop with this

**Review by aapeliv — APPROVED** [review]
LGTM, I'll merge if tests pass!

I thought it'd be more messy and mess up your PRs as well. Thanks for fixing!

**Comment by darrenvong** [comment]
Same, glad they were only small tweaks 😌

**Pull request merged**
